File: packages/native_crypto_android/android/src/main/kotlin/fr/pointcheval/native_crypto_android/interfaces/Cipher.kt
package fr.pointcheval.native_crypto_android.interfaces

import fr.pointcheval.native_crypto_android.utils.FileParameters

interface Cipher {
    fun encrypt(data: ByteArray, key: ByteArray, predefinedIV: ByteArray?): ByteArray
    fun decrypt(data: ByteArray, key: ByteArray): ByteArray
    fun encryptFile(fileParameters: FileParameters, key: ByteArray, predefinedIV: ByteArray?): Boolean
    fun decryptFile(fileParameters: FileParameters, key: ByteArray): Boolean
}

File: packages/native_crypto_ios/ios/Classes/protocols/Cipher.swift
//
//  Cipher.swift
//  native_crypto_ios
//
//  Created by Hugo Pointcheval on 25/05/2022.
//

import Foundation

protocol Cipher {
    func encrypt(data: Data, key: Data, predefinedIV: Data?) throws -> Data
    func decrypt(data: Data, key: Data) throws -> Data
    func encryptFile(fileParameters: FileParameters, key: Data, predefinedIV: Data?) throws -> Bool
    func decryptFile(fileParameters: FileParameters, key: Data) throws -> Bool
}
